pkg/controlplane/instance.go

```diff
@@ -425,7 +425,7 @@ func (c CompletedConfig) StorageProviders(discovery clientdiscovery.DiscoveryInt
 		appsrest.StorageProvider{},
 		admissionregistrationrest.RESTStorageProvider{Authorizer: c.ControlPlane.Generic.Authorization.Authorizer, DiscoveryClient: discovery},
 		eventsrest.RESTStorageProvider{TTL: c.ControlPlane.EventTTL},
-		resourcerest.RESTStorageProvider{},
+		resourcerest.RESTStorageProvider{LoopbackClientConfig: c.ControlPlane.Generic.LoopbackClientConfig},
```

Comment:
we already have a constructed client, change the signature of StorageProviders to take a full client instead of just the discovery client:

```go
	restStorageProviders, err := c.StorageProviders(client)
```

then pass *just* the namespace subclient down into resourcerest.RESTStorageProvider to limit the surface area that storage gets:
```go
resourcerest.RESTStorageProvider{client.CoreV1().Namespaces()}
```